### ISSUE TITLE
The `http.request()` `path` option is not accepted

### ISSUE THREAD

**Issue by lpinca** [sindresorhus/got#973]
#### Describe the bug

 - Node.js version: v13.3.0
 - OS & version: macOS 10.15.1

The documentation says that the `url` argument can be a [`http.request()`](https://nodejs.org/api/http.html#http_http_request_options_callback) options object but the `path` option is not accepted as a valid option.

#### Actual behavior

A `TypeError` error is thrown if the `path` option is used.

#### Expected behavior

The `path` option is accepted and works as documented in https://nodejs.org/api/http.html#http_http_request_options_callback.

#### Code to reproduce

```js
const got = require('got');

const options = {
  hostname: 'example.com',
  method: 'GET',
  path: '/?foo=bar',
  protocol: 'https:'
};

(async function() {
  const { body } = await got(options);
  console.log(body);
})().catch(console.error);
```

#### Checklist

- [x] I have read the documentation.
- [x] I have tried my code with the latest version of Node.js and Got.

**Comment by szmarczak** [comment]
You have to split `path` into `pathname` and `search` as these are URL parameters. Legacy Url support was removed in Got 10, hence the `TypeError`.

I see that is a big breaking change, so I'm gonna fix this.

**Comment by lpinca** [comment]
> You have to split `path` into `pathname` and `search` as these are URL parameters.

That's what I'm doing as workaround but documentation is misleading. The same happens with the `auth` options. If `path` and `auth` are not supported it should be documented.

**Comment by szmarczak** [comment]
Fixed in b3f1ac9
